Liquibase throws exception on `update()` after migration to Spring Boot 2.6.1

After migration to Spring Boot 2.6.1 the following piece of code
```java
var database = (MongoLiquibaseDatabase) DatabaseFactory.getInstance().openDatabase(url, null, null, null, null);
var liquibase = new Liquibase(liquibaseProperties.getChangeLog(), new ClassLoaderResourceAccessor(), database);
liquibase.update("");
```
throws exception
```
org.bson.codecs.configuration.CodecConfigurationException: Can't find a codec for interface java.util.List.
liquibase.exception.LockException: org.bson.codecs.configuration.CodecConfigurationException: Can't find a codec for interface java.util.List.
	at app//liquibase.nosql.lockservice.AbstractNoSqlLockService.acquireLock(AbstractNoSqlLockService.java:183)
	at app//liquibase.nosql.lockservice.AbstractNoSqlLockService.waitForLock(AbstractNoSqlLockService.java:119)
	at app//liquibase.Liquibase.lambda$update$1(Liquibase.java:214)
	at app//liquibase.Scope.lambda$child$0(Scope.java:177)
	at app//liquibase.Scope.child(Scope.java:186)
	at app//liquibase.Scope.child(Scope.java:176)
	at app//liquibase.Scope.child(Scope.java:155)
	at app//liquibase.Liquibase.runInScope(Liquibase.java:2404)
	at app//liquibase.Liquibase.update(Liquibase.java:211)
	at app//liquibase.Liquibase.update(Liquibase.java:197)
	at app//liquibase.Liquibase.update(Liquibase.java:193)
	at app//liquibase.Liquibase.update(Liquibase.java:185)
        ...
Caused by: org.bson.codecs.configuration.CodecConfigurationException: Can't find a codec for interface java.util.List.
	at app//org.bson.codecs.BsonTypeCodecMap.get(BsonTypeCodecMap.java:68)
	at app//org.bson.codecs.ContainerCodecHelper.readValue(ContainerCodecHelper.java:41)
	at app//org.bson.codecs.DocumentCodec.decode(DocumentCodec.java:177)
	at app//org.bson.codecs.DocumentCodec.decode(DocumentCodec.java:44)
	at app//org.bson.internal.LazyCodec.decode(LazyCodec.java:48)
	at app//org.bson.codecs.ContainerCodecHelper.readValue(ContainerCodecHelper.java:61)
	at app//org.bson.codecs.DocumentCodec.decode(DocumentCodec.java:177)
	at app//org.bson.codecs.DocumentCodec.decode(DocumentCodec.java:44)
	at app//org.bson.codecs.ContainerCodecHelper.readValue(ContainerCodecHelper.java:61)
	at app//org.bson.codecs.DocumentCodec.decode(DocumentCodec.java:177)
	at app//org.bson.codecs.DocumentCodec.decode(DocumentCodec.java:44)
	at app//org.bson.Document.parse(Document.java:126)
	at app//liquibase.ext.mongodb.statement.BsonUtils.lambda$orEmptyDocument$0(BsonUtils.java:69)
	at java.base@17.0.1/java.util.Optional.map(Optional.java:260)
	at app//liquibase.ext.mongodb.statement.BsonUtils.orEmptyDocument(BsonUtils.java:69)
	at app//liquibase.ext.mongodb.statement.RunCommandStatement.<init>(RunCommandStatement.java:30)
	at app//liquibase.ext.mongodb.lockservice.AdjustChangeLogLockCollectionStatement.<init>(AdjustChangeLogLockCollectionStatement.java:40)
	at app//liquibase.ext.mongodb.lockservice.MongoLockService.adjustRepository(MongoLockService.java:101)
	at app//liquibase.nosql.lockservice.AbstractNoSqlLockService.init(AbstractNoSqlLockService.java:102)
	at app//liquibase.nosql.lockservice.AbstractNoSqlLockService.acquireLock(AbstractNoSqlLockService.java:155)
	... 97 more
```
However, the same code works fine with older version like 2.5.3

After some investigation, I figured out Spring Boot 2.6.1 uses `4.4.0` version of mongodb driver while `liquibase-mongodb:4.6.2` uses 4.3.4. Seems like something has changed in `Document.parse(...)` in new driver version so that [this](https://github.com/liquibase/liquibase-mongodb/blob/main/src/main/java/liquibase/ext/mongodb/statement/BsonUtils.java#L69) line throws exception with provided `DocumentCodec`. This exception could be reproduced by calling this snippet
```java
        Document.parse("""
                { collMod: "DATABASECHANGELOGLOCK",
                "validator": {
                     "$jsonSchema": {
                         "bsonType": "object",
                         "description": "Database Lock Collection",
                         "required": ["_id", "locked"],
                             "properties": {
                                 "_id": {
                                     "bsonType": "int",
                                     "description": "Unique lock identifier"
                                 },
                                 "locked": {
                                     "bsonType": "bool",
                                     "description": "Lock flag"
                                 },
                                 "lockGranted": {
                                     "bsonType": "date",
                                     "description": "Timestamp when lock acquired"
                                 },
                                 "lockedBy": {
                                     "bsonType": ["string", "null"],
                                     "description": "Owner of the lock"
                                 }
                             }
                         }
                     },
                "validationLevel": "strict",
                "validationAction": "error"
                }""", BsonUtils.DOCUMENT_CODEC)
```
while this one without `BsonUtils.DOCUMENT_CODEC` works fine
```java
        Document.parse("""
                { collMod: "DATABASECHANGELOGLOCK",
                "validator": {
                     "$jsonSchema": {
                         "bsonType": "object",
                         "description": "Database Lock Collection",
                         "required": ["_id", "locked"],
                             "properties": {
                                 "_id": {
                                     "bsonType": "int",
                                     "description": "Unique lock identifier"
                                 },
                                 "locked": {
                                     "bsonType": "bool",
                                     "description": "Lock flag"
                                 },
                                 "lockGranted": {
                                     "bsonType": "date",
                                     "description": "Timestamp when lock acquired"
                                 },
                                 "lockedBy": {
                                     "bsonType": ["string", "null"],
                                     "description": "Owner of the lock"
                                 }
                             }
                         }
                     },
                "validationLevel": "strict",
                "validationAction": "error"
                }""")
```

--- a/src/main/java/liquibase/ext/mongodb/statement/BsonUtils.java
+++ b/src/main/java/liquibase/ext/mongodb/statement/BsonUtils.java
@@ -25,12 +25,7 @@ import com.mongodb.MongoClientSettings;
 import lombok.NoArgsConstructor;
 import org.bson.Document;
 import org.bson.UuidRepresentation;
-import org.bson.codecs.BsonValueCodecProvider;
-import org.bson.codecs.DocumentCodec;
-import org.bson.codecs.DocumentCodecProvider;
-import org.bson.codecs.UuidCodec;
-import org.bson.codecs.UuidCodecProvider;
-import org.bson.codecs.ValueCodecProvider;
+import org.bson.codecs.*;
 import org.bson.codecs.configuration.CodecRegistries;
 import org.bson.codecs.configuration.CodecRegistry;
 
@@ -52,7 +47,8 @@ public final class BsonUtils {
                     new ValueCodecProvider(),
                     new BsonValueCodecProvider(),
                     new DocumentCodecProvider(),
-                    new DBRefCodecProvider()));
+                    new DBRefCodecProvider(),
+                    new IterableCodecProvider()));
 
     public static final String ITEMS = "items";
 

